src/cpp/session/modules/SessionRSConnect.R

```diff
@@ -204,26 +204,37 @@
 })
 
 .rs.addJsonRpcHandler("get_rsconnect_app", function(id, account, server, hostUrl) {
-   # init with empty list
-   appList <- list()
-   appError <- ""
+   
+   # NOTE: We previously used `rsconnect:::getAppById()`, but this API did not
+   # provide the requisite 'config_url' entry, which we display in UI for redeployments.
+   
+   # collect application list
+   apps <- tryCatch(
+      rsconnect::applications(account = account, server = server),
+      error = identity
+   )
+   
+   if (inherits(apps, "error")) {
+      message <- conditionMessage(apps)
+      return(list(error = .rs.scalar(message), app = NULL))
+   }
+
+   # drop __api__ suffix from hostUrl if necessary
+   hostUrl <- sub("/__api__$", "/", hostUrl)
+   
+   # keep only application records which:
+   # - start with the provided host URL;
+   # - have a matching id
+   apps <- apps[.rs.startsWith(apps$url, hostUrl) & apps$id == id, ]
```

Comment:
Is it necessary to filter by URL? The account/server should have already restricted the result set.

Reply:
Thanks -- looking more closely, it seems like `hostUrl` is used in case the application seemingly isn't available from the provided account + server:

https://github.com/rstudio/rsconnect/blob/7d964a541c61183069d286d3e20f071dee39618e/R/ide.R#L92-L106

but I can't see how that'd ever happen in the way we're using it in RStudio.